modules/opensearch/includes/opensearch.search.inc

```diff
@@ -601,3 +601,46 @@ function opensearch_search_object_ids($ids) {
   }
   return FALSE;
 }
+
+function opensearch_search_autocomplete($query) {
+  $items = array();
+  $rest_url = variable_get('opensearch_search_autocomplete_suggestion_url', '');
+  $method = variable_get('ting_search_autocomplete_method', 'facets');
+  $settings = ting_search_autocomplete_settings();
+
+  if (empty($settings['index'])) {
+    watchdog('ting_search_autocomplete', 'Autocomplete index not defined ', array(), $severity = WATCHDOG_WARNING);
+    drupal_json_output(array());
+    drupal_exit();
```

Comment:
It would probably be more appropriate to return an empty array here and let `ting_search_autocomplete` decide what to do about it. In theory you cannot be sure about the context from which the provider implementation is invoked.